public/js/Lineplot.js

```diff
@@ -209,56 +216,90 @@ Lineplot.prototype.updateActiveLine = function(feature) {
+ * and handles the update.
  *
  * @this {Lineplot}
  * @param {number} keyCode - The Javascript character code for the
  *     pressed key.
  *     See https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode
+ * @private
  */
-Lineplot.prototype.keypressUpdate = function(keyCode) {
-    var self = this;
-
-    if(self.activeFeature) {
-        if(keyCode === RIGHTARROW &&
-            // only let the line move right if it's not already at the
-            // rightmost feature
-            self.activeFeature < this.featureVector.length) {
-            self.activeFeature++;
-            $('body').trigger('updateLineplot', self.activeFeature);
-        }
-        else if(keyCode === LEFTARROW && self.activeFeature > 1) {
-            // only let the line move left if it's not already at the
-            // left-most feature
-            self.activeFeature--;
-            $('body').trigger('updateLineplot', self.activeFeature);
-        }
+Lineplot.prototype._onKeydownUpdate = function(keyCode) {
+    // only let the line move right if it's not already at the
+    // rightmost feature
+    if(keyCode === RIGHTARROW &&
+        this.activeFeature < this.featureVector.length) {
+        this.activeFeature++;
+        this._handleFeatureChange();
+    }
+    // only let the line move left if it's not already at the
+    // left-most feature
+    else if(keyCode === LEFTARROW && this.activeFeature > 1) {
+        this.activeFeature--;
+        this._handleFeatureChange();
     }
 };
 
 /**
- * destroy: Remove all child SVG elements of the plot objects containing div.
+ * setScale: Set scale for current feature vector.
+ *
+ * Sets the scale that maps the data to the plot.
+ *
+ * @private
+ */
+Lineplot.prototype._setScale = function() {
+    // get min/max x/y values -- needed for scaling axis/data
+    var yMin = d3.min(this.featureVector);
+    var yMax = d3.max(this.featureVector);
+
+    // create margin for y axis -- stops the line from
+    // drawing at the very top or bottom of the plot
+    var yMargin = (yMax - yMin) * this.axisMargin;
+
+    this.xScale = d3.scale.linear()
+        .domain([0, this.featureVector.length])
+        .range([0, this.width]);
+
+    this.yScale = d3.scale.linear()
+        .domain([yMin - yMargin, yMax + yMargin])
+        .range([this.height, 0]);
+};
+
+/**
+ * updateActiveLine: Redraw the line overlaying selected feature in the plot.
  *
  * @this {Lineplot}
+ * @private
  */
-Lineplot.prototype.destroy = function() {
-    d3.select(this.element + ' > svg').remove();
+Lineplot.prototype._updateActiveLine = function() {
+    var xCoord = this.xScale(this.activeFeature);
+    this.svg.selectAll('.selectedLine').remove();
+    this.svg.append('line')
+        .attr('x1', xCoord)
+        .attr('y1', 0)
+        .attr('x2', xCoord)
+        .attr('y2', this.height)
+        .attr('stroke', 'red')
+        .attr('stroke-width', 0.75)
+        .attr('stroke-dasharray', '5, 5')
```

Comment:
It's an SVG attribute, they always take strings as their type. Numbers just end up getting parsed as strings anyway :)

eg.
```
<svg height="210" width="500">
  <line x1="0" y1="0" x2="200" y2="200" style="stroke:rgb(255,0,0);stroke-width:2" />
</svg>
```